### ISSUE TITLE
Better initial window sizes

### ISSUE THREAD

**Issue by rcoreilly** [goki/gi#381]
The Dialog code has special logic to precompute the size of a window based on contents, and then use that for initial window size -- use that more widely for other windows -- have been lazily relying on window size memory but would be better for it get a reasonable size first.

Failing that, could have better initial guesses that are used instead of the same default for everything.

**Comment by rcoreilly** [comment]
much improved -- most cases are prefect-sized, but sliceviewbase (and tableview) cannot as they size the view based on avail size -- so they only allocate post render.

**Comment by HDHChris** [comment]
Hi, I found that setting the width and height of the main form does not take effect. Even if I have never compiled and run the program before, that is to say, even the width and height parameters set during the first rendering of the main window do not take effect. It seems that it always renders and loads the default window size according to the same width and height value.How can I make it render according to the value I set? Thank you. Examples are as follows:
<img width="2221" alt="image" src="https://user-images.githubusercontent.com/76725654/165289095-dd968f33-1851-4640-b2c2-117c3822f742.png">
